docs/getting-started/quickstart-programming.md

```diff
@@ -79,178 +57,232 @@ exports.up = function(knex) {
+
+## Reading from the Database
 
-To learn more about accessing the database in DBOS, see [our guide](../tutorials/transaction-tutorial.md).
+You can add another GET API function to read all the greetings from the database like so:
+
+```javascript
+//export class Greetings {
+//...
+  @Transaction({readOnly: true})
+  @GetApi('/greetings')
+  static async allGreetings(ctxt: TransactionContext<Knex>) {
+    return await ctxt.client('greetings').select('*') as GreetingRecord[];
+  }
+//}
+```
+
+Here we add a new `@GetApi` function that is also a `@Transaction`. We decorate this transaction as `{readOnly: true}` since it does not modify the database. This enables DBOS to execute it faster, with fewer database round-trips. To learn more about accessing the database in DBOS, see [our guide](../tutorials/transaction-tutorial.md).
 
 :::info
-In this quickstart, we write our database operations in raw SQL (using [knex.raw](https://knexjs.org/guide/raw.html)), but DBOS Transact also supports [knex's query builder](https://knexjs.org/guide/query-builder.html), [TypeORM](https://typeorm.io/), and [Prisma](https://www.prisma.io/docs/orm/prisma-client).
+In this quickstart, we run queries using [Knex query builder](https://knexjs.org/guide/query-builder.html) but DBOS Transact also supports [Knex.raw](https://knexjs.org/guide/raw.html), [TypeORM](https://typeorm.io/), and [Prisma](https://www.prisma.io/docs/orm/prisma-client).
 :::
 
-### Interacting with External Services
+## Interacting with External Services
 
-:::info what you will learn
-How to safely send requests to third-party APIs.
-:::
+Let's say we want to use a third-party service to send our greeting, or record it in a remote system. For this example, you can use our demo Guestbook app. It lets you generate a key and use it to record a few temporary greetings.
+
+### Create a Guestbook Key
+To create a new key, visit https://demo-guestbook.cloud.dbos.dev/key It should output a 36-character sequence like so `12345abc-1234-5678-1234-567890abcdef`. Yours will be different. You can pass this key to your app as a config variable. In your app folder, edit the file `dbos-config.yaml`. Add a new `env:` section at the bottom with the variable `GUESTBOOK_KEY` set to your key in quotes:
+```yaml
+env:
+  GUESTBOOK_KEY: 'your-key-value-here'
+```
+
+For example, if your key value is `12345abc-1234-5678-1234-567890abcdef` then you would add:
+```yaml
+env:
+  GUESTBOOK_KEY: '12345abc-1234-5678-1234-567890abcdef'
+```
+::::tip
+In production, to avoid storing keys in cleartext and to make rotation easier you could change this entry to `GUESTBOOK_KEY: ${ENV_GUESTBOOK_KEY}`. You would then set `ENV_GUESTBOOK_KEY` in your enviornment prior to running `npx dbos start`. This also applies to cloud deployment: the variable is expanded at `npx dbos-cloud app deploy` time and securely passed to the cloud.
```

Comment:
Provide a link to `/api-reference/configuration#environment-variables` so we can greatly shorten this paragraph.

Reply:
done. I added a couple words to the configuration section to explain this a little more